src/datacenters/parser.ts

```diff
@@ -1,12 +1,98 @@
+import * as hclParser from 'hcl2-parser';
 import Ajv2019 from 'https://esm.sh/v124/ajv@8.11.0/dist/2019.js';
 import yaml from 'js-yaml';
+import { ResourceTypeList } from '../@resources/types.ts';
 import * as DatacenterSchemaContents from './datacenter-schema.ts';
 import { Datacenter } from './datacenter.ts';
 import { buildDatacenter, DatacenterSchema } from './schema.ts';
 
 const DEFAULT_SCHEMA_VERSION = 'v1';
 const ajv = new Ajv2019({ strict: false, discriminator: true });
 
+const convertAccounts = (obj: any, resultObj: any): any => {
```

Comment:
1. We should try to use generics here instead of `any` for typing protection. 
2. What do these 4 methods do? Just pluralize the keyword?

Reply:
My current local version removes the `any` keyword but does break the versioning system. Though my plan to fix that is to mark the type of object in the payload and then have the `DatacenterV1` do the actual transformation. Unless the Datacenter itself does the work its impossible to not break the version system.

They do a bit more than that.
1. Pluralizes some words.
2. Convert strings like `environment.vpc.vpc.id` to `environment.resources.vpc.id`
3. HCL uses `${}` while we use `${{}}` so we switch them out.
4. HCL uses arrays in a bunch of places we do not. This removes that.
5. Handle minor differences like the fact we do not have a default hook and instead just have `defaults`. We just convert it to a hook and add it to the end.